src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -101,7 +85,7 @@ private void RaiseAnalyzerChangedWarning(ProjectId projectId, string analyzerPat
             }
         }
 
-        internal void AddPath(string filePath)
+        internal void TrackFilePathAndReportErrorIfChanged(string filePath, ProjectId projectId)
```

Comment:
if warning is all we care, (compared to synchronously blocking changes)

can we use a queue to guarantee event ordering? (either SimpleTaskQueue or just Task + continuation)

that way we could reduce/remove race possibility since all event will overlap?

Reply:
@heejaechang I was intentionally keeping that out of scope for now. This will need a larger revisit once we move this to the full free-threaded thing, so we'll just do that then.